src/main/groovy/liquibase/harness/base/BaseCompatibilityTest.groovy

```diff
@@ -0,0 +1,225 @@
+
+        then: "execute metadata checking sql, obtain result set, compare it to expected result set"
+        Connection newConnection
+        JSONArray generatedResultSetArray
+        try {
+            ResultSet resultSet
+            if (connection.isClosed()) {
+                newConnection = DriverManager.getConnection(testInput.url, testInput.username, testInput.password)
+                resultSet = newConnection.createStatement().executeQuery("SELECT * FROM DATABASECHANGELOG")
+                generatedResultSetArray = mapResultSetToJSONArray(resultSet)
+                if (!newConnection.autoCommit) {
+                    newConnection.commit()
+                }
+            } else {
+                resultSet = ((JdbcConnection) connection).createStatement().executeQuery("SELECT * FROM DATABASECHANGELOG")
+                generatedResultSetArray = mapResultSetToJSONArray(resultSet)
+                if (!connection.autoCommit) {
+                    connection.commit()
+                }
+            }
+            def expectedResultSetArray = new JSONObject(expectedResultSet).getJSONArray(testInput.change)
+            assert compareJSONArrays(generatedResultSetArray, expectedResultSetArray, JSONCompareMode.LENIENT)
+        } catch (Exception exception) {
+            Scope.getCurrentScope().getUI().sendMessage("Error executing metadata checking sql! " + exception.printStackTrace())
+            Assert.fail exception.message
+        } finally {
+            if (newConnection != null) {
+                newConnection.close()
+            }
+        }
+
+        and: "check for actual presence of created object"
+        try {
+            executeQuery(pathToXmlCheckingSqlFile, testInput)
+            executeQuery(pathToYamlCheckingSqlFile, testInput)
+            executeQuery(pathToJsonCheckingSqlFile, testInput)
+            executeQuery(pathToSqlCheckingSqlFile, testInput)
+        } catch (SQLException sqlException) {
+            // Assume test object was not created after 'update' command execution and test failed.
+            Scope.getCurrentScope().getUI().sendMessage("Error executing test table checking sql! " +
+                    sqlException.printStackTrace())
+            Assert.fail sqlException.message
+        }
+
+        cleanup: "rollback changes if we ran changeSet"
+        if (shouldRunChangeSet) {
+            argsMap.put("changeLogFile", pathToXmlChangelogFile)
+            strategy.performRollback(argsMap)
+            argsMap.put("changeLogFile", pathToYamlChangelogFile)
+            strategy.performRollback(argsMap)
+            argsMap.put("changeLogFile", pathToJsonChangelogFile)
+            strategy.performRollback(argsMap)
+            argsMap.put("changeLogFile", pathToSqlChangelogFile)
+            strategy.performRollback(argsMap)
+        }
+
+        and: "check for actual absence of the object removed after 'rollback' command execution"
+        if (shouldRunChangeSet) {
+            try {
```

Comment:
in case of using Arrays or Collections to hold 4 values this part should look way better

Reply:
++